### PR TITLE
added job profiles to be selected (preferrably) based on sys.jobs_recent

### PR DESCRIPTION
**Pull request by mxmarg**
- Added sys.jobs_recent to the system tables being collected
- Limited sys.jobs_recent by number of days (instead of number of rows)
- Adapted the logic for CollectJobHistoryJSON to check for the existence of sys.jobs_recent and always use that over queries.json, if possible
- Added QueriesRow object property mapping: job_id -> QueryId ; planner_estimated_cost -> QueryCost; etc.

Open questions:
- parseLineJobsJson is currently not as diligent as parseLine in making sure that keys actually exist in the passed in JSON object
- the following test failed for me locally and may need to be fixed: cmd/local/apicollect/jobprofiles_test.go/TestGetNumberOfJobProfilesCollectedWIthServerUp()

### CODE REVIEW

File: cmd/local/apicollect/jobprofiles.go [hunk 1]
```diff
@@ -35,39 +35,42 @@ import (
...
-			return 0, 0, err
-		}
-		jobhistoryjsons := []string{}
-		for _, file := range files {
-			if strings.Contains(file.Name(), "project.history.jobs") {
-				jobhistoryjsons = append(jobhistoryjsons, path.Join(c.SystemTablesOutDir(), file.Name()))
+	if len(jobhistoryjsons) == 0 {
+
+		// Attempt to read job history from queries.json, if not Dremio Cloud
+		if !c.IsDremioCloud() {
```
Comment (rsvihladremio) [inline cmd/local/apicollect/jobprofiles.go:53]: stupid out loud question..is this even necessary now..basically len(jobhistoryjsons) == 0 will fail consistently for dremiocloud no? Like effectively we will never parse in practice

**Review by rsvihladremio — APPROVED** [review]
we can clean up the extraneous check later..for now this works

File: cmd/local/apicollect/jobprofiles.go (hunk 1 above)
Comment (rsvihladremio) [reply cmd/local/apicollect/jobprofiles.go:53]: also have a failing test 

--- FAIL: TestGetNumberOfJobProfilesCollectedWIthServerUp (0.28s)
    jobprofiles_test.go:98: failed running job profile numbers generation
        open /tmp/TestGetNumberOfJobProfilesCollectedWIthServerUp2146487464/002/system-tables/node1: no such file or directory

Comment (mxmarg) [reply cmd/local/apicollect/jobprofiles.go:53]: The logic now is:
1. Look for Job JSONs that are either from `sys.jobs_recent` (Software) or `sys.project.history.jobs` (Cloud) and collect those job infos
2. If no valid job entries are found, either a) look for queries.json instead (Software) or b) skip job profile collection (Cloud)
2a. If neither Job JSONs or queries.json are found, skip job profile collection (Software)

**Pull request merged**